### DISCUSSION TITLE
QR Code function

### DISCUSSION THREAD

**Discussion by zero80472** [Authenticator-Extension/Authenticator#1178] [General]
Hi , im using this in a large org and wish to understand what permisions are used to  do the QR code scanning as part of the org we cant allow just anyone to beable to scan or open screencapture perms 

Regards
Z

**Comment by Sneezry** [comment]
By default, the extension is granted access only to the active tab upon clicking its icon.

![image](https://github.com/Authenticator-Extension/Authenticator/assets/1621293/3fdc774f-68c8-4ceb-9a62-7faf50dc4371)

When activated, it can capture the region highlighted in green. However, the areas marked in red remain inaccessible and cannot be captured through the extension's icon click.

![Untitled-2](https://github.com/Authenticator-Extension/Authenticator/assets/1621293/8ec75ea7-3e31-4a12-9ac1-05ec897cd755)

* Related code: https://github.com/Authenticator-Extension/Authenticator/blob/2145edc9c17b20f530fd559fff958c64ceb6b256/src/background.ts#L62-L129

* `tabs.captureVisibleTab()` API reference: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/captureVisibleTab

**Comment by zero80472** [comment]
this is the same refreance for Chrome;/edge right 

also dose the PNG use file:// to load (not good with js coding)

**Reply by Sneezry** [reply]
> this is the same refreance for Chrome;/edge right

Yes, Chrome, Firefox, and Edge follow the same policy.

> also dose the PNG use file:// to load (not good with js coding)

By default, extensions are not permitted to access the `file://` protocol. However, users can grant extensions the **Allow access to file URLs** permission to enable this access.

![image](https://github.com/Authenticator-Extension/Authenticator/assets/1621293/1b69f7bc-7740-4f8e-9529-5027030eca94)
